kernel/fatal.c

```diff
@@ -110,10 +109,10 @@ void z_fatal_error(unsigned int reason, const struct arch_esf *esf)
 	}
 #endif /* CONFIG_ARCH_HAS_NESTED_EXCEPTION_DETECTION */
 
-	LOG_ERR("Current thread: %p (%s)", thread,
-		thread_name_get(thread));
-
-	coredump(reason, esf, thread);
+	if (IS_ENABLED(CONFIG_MULTITHREADING)) {
+		LOG_ERR("Current thread: %p (%s)", thread, thread_name_get(thread));
+		coredump(reason, esf, thread);
```

Comment:
`coredump()` can deal with NULL thread pointer so this should be outside of the if block.

Reply:
done